### PR TITLE
[WIP] Validate more mbids

### PR DESCRIPTION
**Pull request by alastair**
extension to https://github.com/metabrainz/listenbrainz-server/pull/49
https://github.com/metabrainz/listenbrainz-server/commit/2a46238f625b01f4b863e877e608c6f8b7dc07a2#commitcomment-13741298

I want to wait until https://github.com/metabrainz/listenbrainz-server/pull/56 is finished so that we can set up testing and test these validations.

### CODE REVIEW

File: webserver/views/api.py [hunk 1]
```diff
@@ -292,19 +292,25 @@ def _validate_listen(listen):
...
+def _verify_mbid_validity(listen, key):
+    item = listen['track_metadata']['additional_info'].get(key)
+    if item and not is_valid_uuid(item):
+        _log_raise_400('%s MBID format invalid.' % (key, ), listen)
+
+def _verify_multi_mbid_validity(listen, key):
+    items = listen['track_metadata']['additional_info'].get(key, [])
+    for i in items:
+        if not is_valid_uuid(i):
+            _log_raise_400('%s MBID format invalid.' % (key, ), listen)
```
Comment (gentlecat) [inline webserver/views/api.py]: I think some parts of `_verify_mbid_validity` and `_verify_multi_mbid_validity` can be combined.

File: webserver/views/api.py [hunk 2]
```diff
@@ -96,7 +96,7 @@ def get_listens(user_id):
     Get listens for user ``user_id``. The format for the JSON returned is defined in our :ref:`json-doc`.
 
     If none of the optional arguments are given, this endpoint will return the :data:`~webserver.views.api.DEFAULT_ITEMS_PER_GET` most recent listens.
-    The optional ``max_ts`` and ``min_ts`` UNIX epoch timestamps control at which point in time to start returning listens. You may specify max_ts or 
+    The optional ``max_ts`` and ``min_ts`` UNIX epoch timestamps control at which point in time to start returning listens. You may specify max_ts or
```
Comment (gentlecat) [inline webserver/views/api.py]: Can you split this into multiple rows?

File: webserver/views/api.py (hunk 1 above)
Comment (alastair) [inline webserver/views/api.py]: yes, good idea

File: webserver/views/api.py (hunk 2 above)
Comment (alastair) [inline webserver/views/api.py]: this change is only here because I stripped spaces on save :(
talk to @mayhem

File: webserver/views/api.py [hunk 3]
```diff
@@ -292,19 +292,25 @@ def _validate_listen(listen):
...
-            if not is_valid_uuid(cmbid):
-                _log_raise_400("Recording MBID format invalid.", listen)
-        ambids = listen['track_metadata']['additional_info'].get('artist_mbids', [])
-        for ambid in ambids:
-            if not is_valid_uuid(ambid):
-                _log_raise_400("Artist MBID format invalid.", listen)
+        items = ['release_mbid', 'recording_mbid', 'release_group_mbid', 'track_mbid']
+        for i in items:
+            _verify_mbid_validity(listen, i)
+        multi_items['artist_mbids', 'work_mbids']
```
Comment (gentlecat) [inline webserver/views/api.py]: What does that do?

File: webserver/views/api.py (hunk 2 above)
Comment (mayhem) [inline webserver/views/api.py]: I was waiting for that. :)

Comment (gentlecat) [inline webserver/views/api.py]: OK, never mind then.

**Pull request closed without merging**